### DISCUSSION TITLE
Is it possible to create a JSON formula to extract data from a JSON field?

### DISCUSSION THREAD

**Discussion by xmontero** [nocodb/nocodb#6675] [General]
Suppose one field is of type JSON and has this data:

    {
        "formData": {
            "firstName": "John",
            "lastName": "Doe",
            "birthDate": "1984-03-03"
        },
        "metadata": { "some other": "irrelevant stuff " }
    }

Then I want a colum that represents the name, another the last name and another the birthdate.

In MySQL I usually achieve this on-the-fly recalculation with with a generated field:

    `cachedLeadName` VARCHAR(200) AS (json_unquote(json_extract(`contentJson`,'$.formData.firstName'))) stored,

each time the field `contentJson` changes, the field `cachedLeadName` reflects the change.

Can I do this with NocoDB?

**Comment by dstala** [comment]
Using custom formula's over JSON is not supported yet. If you have it working in MySQL, you can connect to that database as an external source & get benefits of a clean UI on top for your base.

To know more about connecting to external data source - [read here](https://docs.nocodb.com/data-sources/connect-to-data-source)

**Comment by xmontero** [comment]
Thanks, will try. Closing.
